src/lib/workouts/pushup.ts

```diff
@@ -4,40 +4,104 @@ import Workout from '.';
 export default class Pushup extends Workout {
 	private isUp: boolean;
 	private pushups: number;
+	private average_left_wrist: { x: number; y: number } | null;
+	private average_right_wrist: { x: number; y: number } | null;
+	private iterations: number;
 	constructor() {
 		super();
 		this.isUp = true;
+		this.iterations = 0;
 		this.pushups = 0;
+		this.average_left_wrist = null;
+		this.average_right_wrist = null;
 	}
 	public onFrame(poses: Pose[]): string {
+		this.iterations++;
 		const keypoints = poses[0].keypoints;
-		const left_shoulder = keypoints.find((keypoint) => keypoint.name === 'left_shoulder');
-		const right_shoulder = keypoints.find((keypoint) => keypoint.name === 'right_shoulder');
-		const left_elbow = keypoints.find((keypoint) => keypoint.name === 'left_elbow');
-		const right_elbow = keypoints.find((keypoint) => keypoint.name === 'right_elbow');
-		const left_hip = keypoints.find((keypoint) => keypoint.name === 'left_hip');
-		const right_hip = keypoints.find((keypoint) => keypoint.name === 'right_hip');
+		const left_shoulder = keypoints.find((keypoint) => keypoint.name === 'left_shoulder')!;
+		const right_shoulder = keypoints.find((keypoint) => keypoint.name === 'right_shoulder')!;
+		const left_elbow = keypoints.find((keypoint) => keypoint.name === 'left_elbow')!;
+		const right_elbow = keypoints.find((keypoint) => keypoint.name === 'right_elbow')!;
+		const left_wrist = keypoints.find((keypoint) => keypoint.name === 'left_wrist')!;
+		const right_wrist = keypoints.find((keypoint) => keypoint.name === 'right_wrist')!;
 
-		const left_shoulder_to_elbow = Math.round(
-			Math.sqrt((left_shoulder.x - left_elbow.x) ** 2 + (left_shoulder.y - left_elbow.y) ** 2)
+		// keep a running average
```

Comment:
What if left_wrist and/or right_wrist are not in frame?

Reply:
They must be in the frame